appinventor/components/src/com/google/appinventor/components/runtime/Spreadsheet.java

```diff
@@ -655,12 +658,109 @@ public void run() {
    * values on the table have been updated.
    */
   @SimpleEvent(
-    description="The callback event for the WriteRow block, called after the " +
-      "values on the table have finished updating")
-  public void FinishedWriteRow () {
+          description = "The callback event for the WriteRow block, called after the " +
+                  "values on the table have finished updating")
+  public void FinishedWriteRow() {
     EventDispatcher.dispatchEvent(this, "FinishedWriteRow");
   }
 
+  @SimpleFunction(description = "Adds a new sheet inside the Spreadsheet")
+  public void AddSheet(final String sheetName) {
+    if (spreadsheetID == null || spreadsheetID.isEmpty()) {
+      ErrorOccurred("AddSheet: " + "SpreadsheetID is empty.");
+      return;
+    } else if (credentialsPath == null || credentialsPath.isEmpty()) {
+      ErrorOccurred("AddSheet: " + "Credentials JSON is required.");
+      return;
+    }
+    // Run the API call asynchronously
+    // Run the API call asynchronously
+    AsynchUtil.runAsynchronously(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          Sheets sheetsService = getSheetsService();
+          AddSheetRequest addSheetRequest = new AddSheetRequest()
+                  .setProperties(new SheetProperties()
+                          //set title of new sheet as user parameter
+                          .setTitle(sheetName)
+                  );
+          List<Request> requests = new ArrayList<>();
+          requests.add(new Request().setAddSheet(addSheetRequest));
+          BatchUpdateSpreadsheetRequest body = new BatchUpdateSpreadsheetRequest()
+                  .setRequests(requests);
+          sheetsService.spreadsheets().batchUpdate(spreadsheetID, body).execute();
+          // Run the callback event block
+          activity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+              FinishedAddSheet(sheetName);
+            }
+          });
+
+        } catch (Exception e) {
+          e.printStackTrace();
+          ErrorOccurred("AddSheet: " + e.getMessage());
```

Comment:
Note that because the code at this point is running on a background thread you will need to call `activity.runOnUiThread()` with a Runnable that will call `ErrorOccurred`. Otherwise the error blocks will run on the wrong thread.